From c6be5a4b7e4de9be73a2a2505171c73eb4d6d472 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:51:34 -0400
Subject: [tools] perf: Do not synthesize the treads of default guest

Message-id: <1408953210-26343-207-git-send-email-jolsa@redhat.com>
Patchwork-id: 88926
O-Subject: [PATCH RHEL7.1 BZ1133083 206/322] perf tools: Do not synthesize the treads of default guest.
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit c239c25a82808ad2cbc795b69e621c4c24f154d6
Author: Dongsheng Yang <yangds.fnst@cn.fujitsu.com>
Date: Fri Dec 20 15:53:00 2013 -0500

description
===========
As the default guest is designed to handle orphan kernel symboles with
--guestkallsysms and --guestmodules, it has no user space.

So we should skip synthesizing threads if machine is default guest.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/event.c b/tools/perf/util/event.c
index 2905771..45a76c6 100644
--- a/tools/perf/util/event.c
+++ b/tools/perf/util/event.c
@@ -133,6 +133,9 @@ static pid_t perf_event__synthesize_comm(struct perf_tool *tool,
   goto out;
  }
 
+ if (machine__is_default_guest(machine))
+  return 0;
+
  snprintf(filename, sizeof(filename), "%s/proc/%d/task",
    machine->root_dir, pid);
 
@@ -183,6 +186,9 @@ static int perf_event__synthesize_mmap_events(struct perf_tool *tool,
  FILE *fp;
  int rc = 0;
 
+ if (machine__is_default_guest(machine))
+  return 0;
+
  snprintf(filename, sizeof(filename), "%s/proc/%d/maps",
    machine->root_dir, pid);
 
@@ -409,6 +415,9 @@ int perf_event__synthesize_threads(struct perf_tool *tool,
  if (mmap_event == NULL)
   goto out_free_comm;
 
+ if (machine__is_default_guest(machine))
+  return 0;
+
  snprintf(proc_path, sizeof(proc_path), "%s/proc", machine->root_dir);
  proc = opendir(proc_path);
 
-- 
1.7.1